From c3ff8d3f7281d613b392e6b8574da5b7cd7275b6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Bu=CC=88nemann?= <felix.buenemann@gmail.com>
Date: Sun, 22 Nov 2020 23:06:06 +0100
Subject: [PATCH] Add darwin64-arm64-cc build target

This adds support for the Apple Silicon M1 hardware on macOS Big Sur.

CLA: trivial
---
 Configurations/10-main.conf | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index eb92c24..cea4feb 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -1557,6 +1557,14 @@ my %targets = (
         bn_ops           => "SIXTY_FOUR_BIT_LONG",
         perlasm_scheme   => "macosx",
     },
+    "darwin64-arm64-cc" => {
+        inherit_from     => [ "darwin-common", asm("aarch64_asm") ],
+        CFLAGS           => add("-Wall"),
+        cflags           => add("-arch arm64"),
+        lib_cppflags     => add("-DL_ENDIAN"),
+        bn_ops           => "SIXTY_FOUR_BIT_LONG",
+        perlasm_scheme   => "ios64",
+    },
 
 ##### GNU Hurd
     "hurd-x86" => {
-- 
2.23.0

